### DISCUSSION TITLE
How to add a new language?

### DISCUSSION THREAD

**Discussion by ukraine-sdg** [open-sdg/open-sdg#1437] [Q&A]
Hi, could you, please, help to add a Ukrainian version of the sdg web page.
I have already added all the translations on Weblate.

Then I tried to add Ukrainian in config_data.yml in the way which is written in the tutorial. 
I tried to add "uk" and it did not work, then I tried to add "ua" and it did nor work too 
(when I tried to add a pull request, I got "All checks have failed").

Could you, please write how can I add Ukrainian? which steps should I do?

**Comment by LucyGwilliamAdmin** [comment]
@ukraine-sdg take a look at the follow instructions:

1. Point at where sdg-translations are coming from here: https://github.com/ukraine-sdg/data/blob/develop/config_data.yml#L73-L74 - If using open-sdg/sdg-translations, it looks like you will want to point at 1.6.0
2. Then add `uk` to languages in config_data.yml in data repo and _data/site_config.yml in site repo

There are more instructions but firstly a couple of questions:
1. Is English or Ukraine being first/default language?
2. Do you use git to make changes?

**Reply by ukraine-sdg** [reply]
Thank you, Lucy,

I changed "es" to "uk" in  config_data.yml and pointed 1.6.0,  in data repo.
Then I changed   "es" to "uk"  in _data/site_config.yml in site repo

When I tried to create a pull request, "All checks have failed"
Could you, please, help what was wrong?

**Reply by LucyGwilliamAdmin** [reply]
@ukraine-sdg you will need to make the config changes and the folder name changes in the same pull request

**Reply by ukraine-sdg** [reply]
Thank you, Lucy,
it worked.

However, additinal questions arised
Some things are not displayed correctry.
(1) the pictograms for goals are not displayed
(2) when I use English version and want to choose Ukrainian the there in only a list of "English" and " " in the menu

How to fix it?

**Comment by WohnungNDS** [comment]
@LucyGwilliamAdmin @ukraine-sdg 

In the past i had also problems with this translation issue. Sometimes it tooks a day before the translation was shown correctly. I decided to have my translations in my [data repo](https://github.com/WohnungNDS/IM_Data/tree/develop/translations) that makes it in my special case faster.... But my case is very special.. I use open-sdg for the migration report in lower saxony and I only have one language... So perhaps its not the typical use.....

**Comment by ukraine-sdg** [comment]
Thank you, Lucy

Is English or Ukraine being first/default language?

I would like Ukrainian to be the first version, if possible, and English - the second
I have already create the English version

Do you use git to make changes? No, I did not

**Comment by LucyGwilliamAdmin** [comment]
Ok so in the languages fields, you will need to put uk first and then en second.

Also you will need to change the name of these folders to en (instead of es):
- https://github.com/ukraine-sdg/site/tree/develop/_pages/es
- https://github.com/ukraine-sdg/site/tree/develop/_posts/es
